sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp

```diff
@@ -63,10 +63,25 @@ template <typename Compare = std::less<>> class default_sorter {
   void operator()([[maybe_unused]] Group g, [[maybe_unused]] Ptr first,
                   [[maybe_unused]] Ptr last) {
 #ifdef __SYCL_DEVICE_ONLY__
+    // Adjust the scratch pointer based on alignment of the type T.
     // Per extension specification if scratch size is less than the value
-    // returned by memory_required then behavior is undefined, so we don't check
-    // that the scratch size statisfies the requirement.
-    sycl::detail::merge_sort(g, first, last - first, comp, scratch.data());
+    // returned by memory_required, behavior is undefined. So, if std::align
+    // returns nullptr (which means that we don't have enough space) then exit
+    // early.
```

Comment:
I think we can just ignore and continue executing. Seg. fault accessing `nullptr` would be fine.

Reply:
Sounds good, updated.